frontend/src/components/site/SiteProducts.vue

```diff
@@ -120,6 +132,18 @@ const pidOptions = computed(() => {
   }));
 });
 
+const modelOptions = computed(() => {
+  const { value: dataStatusValue } = dataStatus;
+  const { value: selectedProductIdValue } = selectedProductId;
+  if (!dataStatusValue || !selectedProductIdValue || selectedProductIdValue !== "model") {
```

Comment:
```suggestion
  if (!dataStatusValue || selectedProductIdValue !== "model") {
```

Voi myös miettiä onko `dataStatusValue` parempi kuin `dataStatus.value`. Katso myös `pidOptions` ylempänä.